File: solutions/elixir/resistor-color-duo/1/lib/resistor_color_duo.ex
defmodule ResistorColorDuo do
  @doc """
  Calculate a resistance value from two colors
  """
  @spec value(colors :: [atom]) :: integer
  def value([first, second |_]) do
    10 * band_value(first) + band_value(second)
  end

  defp band_value(color) do
    case color do
      :black -> 0
      :brown -> 1
      :red -> 2
      :orange -> 3
      :yellow -> 4
      :green -> 5
      :blue -> 6
      :violet -> 7
      :grey -> 8
      :white -> 9
    end
  end
end

File: solutions/go/resistor-color-duo/1/resistor_color_duo.go
package resistorcolorduo

var encoding = map[string]int{"black": 0, "brown": 1, "red": 2, "orange": 3, "yellow": 4,
                             "green": 5, "blue": 6, "violet": 7, "grey": 8, "white": 9}

// Value should return the resistance value of a resistor with a given colors.
func Value(colors []string) int {
	return 10 * encoding[colors[0]] + encoding[colors[1]]
}
